### DISCUSSION TITLE
api with pagination, why occur database query with limit 1000

### DISCUSSION THREAD

**Discussion by flyly0755** [encode/django-rest-framework#8523] [Question & Answer] [answered]
<img width="820" alt="image" src="https://user-images.githubusercontent.com/45039060/173209991-241fef8b-aefe-4756-9840-7e3dd1845e9c.png">
<img width="533" alt="image" src="https://user-images.githubusercontent.com/45039060/173209996-9079bd33-8c71-4c7a-9d37-0a642aa1ab1c.png">
with this api http://127.0.0.1:8123/api/regioninfo/?page=3 ,there are 3 database query show as debug_toolbar:

1. SELECT COUNT(*) AS "__count" FROM "regioninfo_region" get total number for knowing how many pages these are.
2. SELECT "regioninfo_region"."id",       
      "regioninfo_region"."name",       
      "regioninfo_region"."name2",       
      "regioninfo_region"."pinyin",       
      "regioninfo_region"."parent_id",       
      "regioninfo_region"."deep",       
      "regioninfo_region"."ext_id",       
      "regioninfo_region"."remark"  
      FROM "regioninfo_region" LIMIT 50 OFFSET 100  
this is also can be understand,  one api query get back 50 data.

3. SELECT "regioninfo_region"."id",
       "regioninfo_region"."name",
       "regioninfo_region"."name2",
       "regioninfo_region"."pinyin",
       "regioninfo_region"."parent_id",
       "regioninfo_region"."deep",
       "regioninfo_region"."ext_id",
       "regioninfo_region"."remark"
  FROM "regioninfo_region" LIMIT 1000
how about this query, why with same query as step 2 except limit condition?

**Comment by hudy0** [comment]
Make sure that you are running your SQL in the same thread that handled the request.

The Django debug toolbar only seems to take a look at the SQL statements that are run in the current thread and assumes that these are the only ones that are related to the request that was handled.

**Reply by flyly0755** [reply]
yes, all these 3 sql query are in the same thread related with one same api call.

**Comment by lovelydinosaur** [comment, marked as answer]
When you're using the browsable API you may have other database queries that are required in order to generate the form controls.

I [expect there's a select dropdown on that page](https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/docs/api-guide/settings.md#html-select-field-cutoffs), which is rendering a choice of regions to select from.

Issuing a request to that endpoint using JSON may not make the same number of queries.

**Reply by flyly0755** [reply]
Yes，you are right👍 but how i change the limit number 1000? for example, change it to 100 for reducing database stress.
